cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/realtime/source/KafkaSource.java

```diff
@@ -110,14 +112,14 @@ public void initialize(RealtimeContext context) throws Exception {
   @Nullable
   @Override
   @SuppressWarnings("unchecked")
-  public SourceState poll(Emitter<StructuredRecord> writer, SourceState currentState) {
+  public SourceState poll(Emitter<StructuredRecord> writer, SourceState currentState) throws InterruptedException {
     try {
       // Lets set the internal offset store
       kafkaConsumer.saveState(currentState);
-
       kafkaConsumer.pollMessages(writer);
     } catch (Throwable t) {
       LOG.error("Error encountered during poll to get message for Kafka source.", t);
+      TimeUnit.SECONDS.sleep(EXCEPTION_SLEEP_IN_SEC);
```

Comment:
in addition to this can we keep a flag that indicates whether to log an error or not.  Then the first time we catch and error, we log saying there is some problem, but don't log again until we are able to successfully poll